app/src/main/kotlin/com/exallium/todoapp/editnote/EditNotePresenter.kt

```diff
@@ -66,7 +80,27 @@ class EditNotePresenter(view: EditNoteView,
         }).addToComposite()
     }
 
-    fun setupSaveNoteSubscription(noteId: String) {
+    fun setupSaveNewNoteSubscription() {
+        view.saveNoteClicks()
+                .flatMap { model.saveNote(buildNote()).toObservable() }
+                .subscribe(object : Subscriber<Unit>() {
```

Comment:
similar to this?

`    fun setupTextChangedSubscription() {
        setupTextChangedValidation(view.titleTextChanges(), model::validateNoteTitleText, view::showInvalidNoteTitleError)
        setupTextChangedValidation(view.bodyTextChanges(), model::validateNoteBodyText, view::showInvalidNoteBodyError)
    }

    fun setupTextChangedValidation(textChangedObservable: Observable<CharSequence>,
                                   validationFn: (String) -> Boolean,
                                   showErrFn: () -> Unit) {`?